components/log_wrappers/src/lib.rs

```diff
@@ -88,6 +88,29 @@ fn test_debug() {
     assert_eq!(&buffer.as_string(), "TIME INFO foo, bar: None\n");
 }
 
+pub struct Key<'a>(pub &'a [u8]);
+
+impl<'a> ::slog::Value for Key<'a> {
```

Comment:
You can implement `fmt::Display` like this: https://github.com/tikv/tikv/pull/4087/files#diff-0d0baa1292b5280670393b77e8fb09ecR226

Reply:
@huachaohuang  I don't think we can implement a foreign trait `fmt::Display` for a foreign type `&[u8]`.

Reply:
ok

Reply:
@breeswish But I can add `Display` for `Key` here?

Reply:
I think it's better not to do it. We will have to wrap `&[u8]` to `log_wrapper::Key`, then convert it to `log_wrapper::DisplayValue`. Not as easy to use as it is now. cc @huachaohuang

Reply:
Yes, let's implement `slog::Value` instead of `Display`, because `Option<T: slog::Value>` implements `slog::Value` but `Option<Display>` doesn't implement `Display` or `slog::Value` (in this case, we need additionally wrap `DisplayValue` so that `slog::Value` is implemented for `Option<T: Display>`)